tests/interfaces/legacy_devices_integration/test_autograd_legacy.py

```diff
@@ -196,23 +197,15 @@ def test_no_batch_transform(self, mocker):
             qml.CNOT(wires=[0, 1])
             qml.expval(H)
 
-        tape = qml.tape.QuantumScript.from_queue(q)
+        tape = qml.tape.QuantumScript.from_queue(q, shots=100000)
         spy = mocker.spy(dev.target_device, "batch_transform")
 
-        if not qml.operation.active_new_opmath():
-            with pytest.raises(AssertionError, match="Hamiltonian must be used with shots=None"):
```

Comment:
Was this warning just being raised because the shots on the tape didn't match the shots on the device, or did something change?

Reply:
Previously, if finite shots, device batch transform will be performed, and the `Hamiltonian` would be split. However, if `device_batch_transform=False`, this will not happen, and the `Hamiltonian` gets sent to the device, and the error is raised by the device saying that the `Hamiltonian` isn't supported with finite shots. Now we've deprecated `device_batch_transform` and made it do nothing, device batch transform should always happen, and the `Hamiltonian` should never reach the device, and the warning isn't raised.